src/event/views/CreateEventPage.vue

```diff
@@ -131,24 +154,58 @@ export default defineComponent({
     DatePicker,
     DayPicker,
     TimePicker,
-    TextInput
+    TextInput,
+    LoginPage
   },
-  data() {
-    return {
-      isHidden: true,
-      isDatePickerEvent: true,
-      startTime: set(new Date(), { hours: 9, minutes: 0 }),
-      endTime: set(new Date(), { hours: 21, minutes: 0 }),
-      timezone: "America/Vancouver",
-      eventTitle: "",
-      eventDescription: "",
-      isChecked: true
-    };
+  data: function() {
+    return initialState();
   },
-  computed: {},
   methods: {
     toggleEventType(toggleState: boolean) {
       this.isDatePickerEvent = toggleState;
+    },
+    showModal() {
+      this.isModalVisible = true;
+    },
+    closeModal() {
+      this.isModalVisible = false;
+    },
+    reset() {
+      Object.assign(this.$data, initialState());
+    },
+    async createEvent(e: any) {
+      e.preventDefault();
```

Comment:
what default behaviour is this stopping? Just curious and too lazy to try it myself :P

Reply:
the default form event I believe. If you don't do this the form will hard refresh the page and make a POST request to the current url with the form fields as a query.